web/src/pages/Settings/Config/ConfigPanels/ConfigTemplates/ConfigModal.jsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import Dialog from "@mui/material/Dialog/Dialog";
+import DialogTitle from "@mui/material/DialogTitle/DialogTitle";
+import DialogContent from "@mui/material/DialogContent/DialogContent";
+import Button from "@mui/material/Button/Button";
+import PropTypes from "prop-types";
+
+export default function ConfigModal(props) {
+    const {
+        body,
+        confirm,
+        open,
+        setOpen,
```

Comment:
nit: I would probably rename this something like `onApply`. someone might wanna do something more than simply closing the modal here. it might also obviate `confirm()` too; looks like there's no calling one without the other. my gut feeling is that in this case flexibility is handy versus being too rigorous.